src/nccl_ofi_net.c

```diff
@@ -1580,30 +1579,25 @@ static ncclResult_t ofi_accept(void *listenComm, void **recvComm)
 			goto exit;
 	}
 
-	/* Get local EP address to detect requests to self */
-	ret = fi_getname(&(nccl_ofi_component[dev]->ep->fid),
-	                 (void *)&local_ep_addr,
-	                 &namelen);
-	if (ret != 0) {
-		NCCL_OFI_WARN("Call to fi_getname() failed with RC: %d, ERROR: %s",
-		              ret, fi_strerror(-ret));
-		ret = ncclSystemError;
-		goto error;
-	}
-
-	if (memcmp(local_ep_addr, remote_ep_addr, namelen) == 0) {
+	/* non NULL memory buffer means connection was local */
+	if (conn_info.memory_buffer) {
```

Comment:
I am not sure if we really need `memory_buffer` pointer here. Can't we simply send a boolean flag indicating whether it was a local request or targeted to remote?

Reply:
This is a key point of solution. We need to keep memory buffer at sender side valid until the whole transaction completes. For rendezvous providers this means that buffer must be valid until trecv ends. So we allocate buffer in ofi_connect, keep it alive until trecv_complete and free it here. Using it as an indicator just simplifies code.

Reply:
It doesn't enhances readability of the code. We can compare the address to identify if we are local, assert that we have a valid value of `memory_buffer` and then free it. This will prevent us from any bad programming issues in future.

Reply:
we can have connect_to_self boolean as another field in connectio_info, it will be extra duplicate, but will improve readability.